wp-includes/widgets/class-wp-widget-visual-text.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Widget API: WP_Widget_Visual_Text class
+ *
+ * @package WordPress
+ * @subpackage Widgets
+ * @since 4.8.0
+ */
+
+/**
+ * Core class used to implement a Text widget.
+ *
+ * @since 4.8.0
+ *
+ * @see WP_Widget
+ */
+class WP_Widget_Visual_Text extends WP_Widget_Text {
+
+	/**
+	 * Add hoosk for enqueueing assets when registering all widget instances of this widget class.
+	 *
+	 * @since 2.8.0
+	 * @access public
+	 */
+	public function _register() {
+		add_action( 'admin_print_scripts-widgets.php', array( $this, 'enqueue_admin_scripts' ) );
+		add_action( 'customize_controls_print_scripts', array( $this, 'enqueue_admin_scripts' ) );
+
+		parent::_register();
+	}
+
+	/**
+	 * Outputs the content for the current Text widget instance.
+	 *
+	 * @since 4.8.0
+	 * @access public
+	 *
+	 * @param array $args     Display arguments including 'before_title', 'after_title',
+	 *                        'before_widget', and 'after_widget'.
+	 * @param array $instance Settings for the current Text widget instance.
+	 */
+	public function widget( $args, $instance ) {
+
+		/** This filter is documented in wp-includes/widgets/class-wp-widget-pages.php */
+		$title = apply_filters( 'widget_title', empty( $instance['title'] ) ? '' : $instance['title'], $instance, $this->id_base );
+
+		$widget_text = ! empty( $instance['text'] ) ? $instance['text'] : '';
+
+		/**
+		 * Filters the content of the Text widget.
+		 *
+		 * @since 2.3.0
+		 * @since 4.4.0 Added the `$this` parameter.
+		 *
+		 * @param string         $widget_text The widget content.
+		 * @param array          $instance    Array of settings for the current widget.
+		 * @param WP_Widget_Text $this        Current Text widget instance.
+		 */
+		$text = apply_filters( 'widget_text', $widget_text, $instance, $this );
+
+		if ( isset( $instance['filter'] ) ) {
+			if ( 'content' === $instance['filter'] ) {
+
+				/**
+				 * Filters the content of the Text widget to apply changes expected from the visual (TinyMCE) editor.
+				 *
+				 * By default a subset of the_content filters are applied, including wpautop and wptexturize.
+				 *
+				 * @since 4.8.0
+				 *
+				 * @param string         $widget_text The widget content.
+				 * @param array          $instance    Array of settings for the current widget.
+				 * @param WP_Widget_Text $this        Current Text widget instance.
+				 */
+				$text = apply_filters( 'widget_text_content', $widget_text, $instance, $this );
+
+			} elseif ( $instance['filter'] ) {
+				$text = wpautop( $text ); // Back-compat for instances prior to 4.8.
+			}
+		}
+
+		echo $args['before_widget'];
+		if ( ! empty( $title ) ) {
```

Comment:
In the other widgets we return if the main content is empty ( i.e. an image is missing ) - should we not render anything including the `$title` if `$text` is empty?